autobot-slm-backend/ansible/tests/check_ansible_core_pinned.py

```
@@ -0,0 +1,107 @@
+
+def read_pin() -> str:
+    """Return the pinned ansible-core version from the single source."""
+    if not _PIN_FILE.is_file():
+        raise SystemExit(f"MISSING PIN FILE: {_PIN_FILE}")
+
+    for line in _PIN_FILE.read_text(encoding="utf-8").splitlines():
+        match = _PIN_RE.match(line.strip())
+        if match:
+            return match.group(1)
+
+    raise SystemExit(f"NO PIN IN {_PIN_FILE}\n" "  Expected a line of the form 'ansible-core==<version>'.")
```

Ruff (0.16.3)

[warning] Avoid specifying long messages outside the exception class

(TRY003)